plutus-ir/src/Language/PlutusIR/MkPir.hs

```diff
@@ -22,4 +25,4 @@ mkLet
     -> [Binding tyname name uni a]
     -> Term tyname name uni a
     -> Term tyname name uni a
-mkLet x r bs t = if null bs then t else Let x r bs t
+mkLet x r bs t = if null bs then t else Let x r (NE.fromList bs) t
```

Comment:
This is technically safe since you're guarding against the case in which the partial function will throw, but it's not very nice.

Why not:
```
case NE.nonEmpty bs of { Nothing -> t; Just bs -> Let x r bs t; }
```